header_test.go

```diff
@@ -52,7 +52,6 @@ func TestHeaderMerge(t *testing.T) {
 	expect := http.Header{
 		"Foo": []string{"one", "two"},
 		"Bar": []string{"one"},
-		"Baz": nil,
```

Comment:
What caused this change?

Reply:
See the change to `mergeHeaders` above.

And the reason `mergeHeaders` was changed was so that `len(responseTrailers)` tells me the actual number of keys, without having to actually iterate the map and ignore nil/empty keys. This is for deciding if there were any trailers in the request (since a trailers-only response is now defined as "no body, no trailers").